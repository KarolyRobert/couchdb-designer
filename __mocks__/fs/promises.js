
'use strict';

const promises = jest.createMockFromModule('fs/promises');

promises.readFile = jest.fn();
promises.writeFile = jest.fn();
promises.readdir = jest.fn();
promises.access = jest.fn();
promises.stat = jest.fn();

module.exports = promises;
