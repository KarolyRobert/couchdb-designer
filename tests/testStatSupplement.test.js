import testStatSupplement  from '../src/testing/testStatSupplement';
import fs from 'fs/promises';

jest.mock('fs/promises');

const fileStats = { 
    isJSON : false,
     isLib : true, 
     name : "map", 
     filePath: "./appdesign/views/cloud/map.js", 
     isJavaScript: true
    }
const fileStat = { 
    mtimeMs : 1000
}

describe('testStatSupplement',() => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('unmodifyed',() => {
        expect.assertions(1);
        fs.stat.mockResolvedValue({mtimeMs: 1100});
        testStatSupplement(fileStats,fileStat).then(result => {
            expect(result).toMatchSnapshot();
        });
    });

    test('modifyed',() => {
        expect.assertions(1);
        fs.stat.mockResolvedValue({mtimeMs: 900});
        testStatSupplement(fileStats,fileStat).then(result => {
            expect(result).toMatchSnapshot();
        });
    });

    test('modifyed on testStatfile not exist',() => {
        expect.assertions(1);
        fs.stat.mockRejectedValue({code:'ENOENT'});
        testStatSupplement(fileStats,fileStat).then(result => {
            expect(result).toMatchSnapshot();
        });
    });

    test('reject on testStatfile statcheck fail',() => {
        expect.assertions(1);
        fs.stat.mockRejectedValue({code:'SOMEOTHERERROR'});
        testStatSupplement(fileStats,fileStat).catch(result => {
            expect(result).toStrictEqual({code:'SOMEOTHERERROR'});
        });
    });




});