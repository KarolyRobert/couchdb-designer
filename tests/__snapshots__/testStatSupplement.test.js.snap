// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testStatSupplement modifyed 1`] = `
Object {
  "filePath": "./appdesign/views/cloud/map.js",
  "isJSON": false,
  "isJavaScript": true,
  "isLib": true,
  "isModified": true,
  "name": "map",
  "testModule": "../../temp/24ff7500092412df698fc70591ebaabc",
  "testPath": "../../temp/24ff7500092412df698fc70591ebaabc.js",
}
`;

exports[`testStatSupplement modifyed on testStatfile not exist 1`] = `
Object {
  "filePath": "./appdesign/views/cloud/map.js",
  "isJSON": false,
  "isJavaScript": true,
  "isLib": true,
  "isModified": true,
  "name": "map",
  "testModule": "../../temp/24ff7500092412df698fc70591ebaabc",
  "testPath": "../../temp/24ff7500092412df698fc70591ebaabc.js",
}
`;

exports[`testStatSupplement unmodifyed 1`] = `
Object {
  "filePath": "./appdesign/views/cloud/map.js",
  "isJSON": false,
  "isJavaScript": true,
  "isLib": true,
  "isModified": false,
  "name": "map",
  "testModule": "../../temp/24ff7500092412df698fc70591ebaabc",
  "testPath": "../../temp/24ff7500092412df698fc70591ebaabc.js",
}
`;
